HIRS_Utils/src/main/java/hirs/data/persist/certificate/attributes/ComponentClass.java

```diff
@@ -0,0 +1,262 @@
+
+        switch (componentIdentifier) {
+            case OTHER:
+                this.category = NONE_STRING;
+                this.component = OTHER_STRING;
+                break;
+            case UNKNOWN:
+                this.category = NONE_STRING;
+                this.component = UNKNOWN_STRING;
+                break;
+            case ERROR:
+                // Number Format Exception
+                break;
+            default:
+                getCategory(getJsonObject(componentClassPath));
+                break;
+        }
+    }
+
+    /**
+     * Getter for the Category type.
+     *
+     * @return string value of the category
+     */
+    public final String getCategory() {
+        return category;
+    }
+
+    /**
+     * Getter for the Component type.
+     *
+     * @return string value of the component
+     */
+    public final String getComponent() {
+        return component;
+    }
+
+    /**
+     * This is the main way this class will be referenced and how it
+     * will be displayed on the portal.
+     * @return String combination of category and component.
+     */
+    @Override
+    public String toString() {
+        return String.format("%s - %s", category, component);
+    }
+
+    /**
+     * Getter for the JSON Object that is associated with the component value
+     * mapped in the associated JSON file.
+     *
+     * @return a JSON Object
+     */
+    private JsonObject getJsonObject(final Path componentClassPath) {
+        // find the file and load it
+        JsonObject components = null;
+
+        if (Files.notExists(componentClassPath)) {
+            LOGGER.info(String.format("No file found at %s.", componentClassPath.toString()));
+        } else {
+            try {
+                InputStream inputStream = new FileInputStream(componentClassPath.toString());
+                JsonObject object = Json.parse(new InputStreamReader(inputStream,
+                        StandardCharsets.UTF_8)).asObject();
+                components = object.get("Components").asObject();
+            } catch (IOException ex) {
+                // add log file thing here indication issue with JSON File
+                components = null;
+            }
+        }
+
+        return components;
+    }
+
+    /**
+     * Getter for the Category mapped to the associated value in.
+     *
+     * @param categories a JSON object associated with mapped categories in file
+     * @componentIndentifier.
+     */
+    private void getCategory(final JsonObject categories) {
+        int componentID;
+
+        if (categories != null) {
+            for (String name : categories.names()) {
+                componentID = Integer.decode(categories.get(name).asObject().get("ID").asString());
+                // check for the correct flag
+                if ((componentIdentifier & CATEGORY_MASK) == componentID) {
+                    JsonObject subObjects = categories.get(name).asObject().get("Types").asObject();
```

Comment:
Can we name this JsonObject types instead of subObjects? Or something more indicative of what we're storing.

Reply:
Updated the names and also of another variable that was just 'object '.